app/src/androidTest/java/de/test/antennapod/util/service/download/NanoHTTPD.java

```diff
@@ -1156,13 +1156,13 @@ private void decodeMultipartData(String boundary, ByteBuffer fbuf, BufferedReade
                             int offset = stripMultipartHeaders(fbuf, bpositions[boundarycount - 2]);
                             String path = saveTmpFile(fbuf, offset, bpositions[boundarycount - 1] - offset - 4);
                             files.put(pname, path);
-                            value = disposition.get("filename");
-                            value = value.substring(1, value.length() - 1);
+                            value = new StringBuilder(disposition.get("filename"));
+                            value = new StringBuilder(value.substring(1, value.length() - 1));
                             do {
                                 mpline = in.readLine();
                             } while (mpline != null && !mpline.contains(boundary));
```

Comment:
I really wonder what this is for o_O

Reply:
Now the meaning is changed. length() should be called on the filename.